### ISSUE TITLE
Weapon - Large Cartoon Bomb

### ISSUE THREAD

**Issue by maximtsai** [maximtsai/AGDAGame#7]
A disposable weapon that starts ticking down when you press the activate button. Explodes at a random time between 3-5 seconds (random so that experienced players won't be able to time bomb throws perfectly).

Bomb makes a large explosion, specifics of explosion up to implementer.

**Comment by BryanTai** [comment]
I am personally skeptical about the random timer but then again, I've personally never seen it implemented in a game before.

At the very least, the bomb should have some indicator that it's about to explode so users don't feel cheated when they're trying to escape or when their own bomb blows up in their face (see Link's Bomb in Super Smash Bros)

**Comment by maximtsai** [comment]
Indicator for sure. We can first try making the bomb without the random timer and if no issues are encountered then we'll just keep it at that.

**Comment by kbramk** [comment]
Since I'm included in this I'll just add my 2 cents here eh?

Clarity is very important and reduces 'feel bad' moments where the player
using or hit by the bomb gets destroyed and feels there's nothing they
could have done to outplay it because it was random.  I'd even go as far as
to say have very clear visual indicators (think what Smash/Zelda do) or
even as blatant as the number ticking down so you know when it's about to
explode.  Even if the timer is randomly from 3-5 seconds or whatever,
having a very clear visual is important so the bomb feels good.  This is
probably one of those things where you code it, try it at 3 seconds, try it
again at 4 seconds, try it again at 5 seconds, a number in between, some
randomness added, and so on until you get it right.  I' haven't played with
it so I don't know but you'll only know through playtesting to make sure it
feels right and that it counters whatever it's supposed to counter.  Figure
out what it's good against, what it's bad against.  Can a weapon be changed
to better counter or better lose to the bomb?  Does something new need to
be added to fill the role?

in short:  Make it very clear when this fucker explodes.  Playtest the fuck
out of it with different values.

- Bram

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by maximtsai** [comment]
Lol thanks for the input Bram! I do plan on having warning indicators for EVERYTHING that can potentially deal damage (cannons charging animation, lasers warming up indicator, bombs beeping faster as they about to blow, giant warning sign on the ground if we implement aerial strikes etc.). If it can kill you, it'll make sure you and everyone else in a suitable radius knows about it.

And yes, playtesting is absolutely crucial. We showcased the Swordship game at the Vancouver Comic Arts Festival and added a new weapon the night before. Playtesting with random people showed that it was actually stupidly overpowered against everyone except experienced players (which was probably why I didn't catch the potential issue).
